chore: cleanup frappe connect II/II

blaggacao opened the pull request:
follow up on #24428

- [x] check why CI ~~is failing~~

blaggacao committed:
chore: simplify frappe.connect and use site config for bootstrapping

blaggacao committed:
test: fully spec test command line

blaggacao committed:
chore: cleanup frappe.connect invocations

blaggacao committed:
fix: default db_user where config defaults are set

blaggacao committed:
fix: remove db name 'hack'; now the code path is clean

blaggacao committed:
fix: assert minimum contract for frappe.connect

blaggacao commented:
@akhilnarang This is now green as grass. I needed to move your defaulting logic around because it appears that some tests, for reasons I don't understand are otherwise missing this, because the site is already initialized, but regarless, `db_user` wasn't set.

I also continued to re-evaluate calls to `connect` and removed them where they seem to be a legacy, non-documented entry-point, that may have been used during development or long before now.

blaggacao commented on frappe/test_runner.py:
```diff
@@ -38,6 +38,7 @@ def _runner(*args, **kwargs):
 
 
 def main(
+	site=None,
```
following the same pattern as parallel_test_runner, now.

akhilnarang commented on frappe/__init__.py:
```diff
@@ -387,6 +394,9 @@ def db_default_ports(db_type):
 	)
 
+	# Set the user as database name if not set in config
+	config["db_user"] = config.get("db_user") or config.get("db_name")
```
May as well follow the rest and allow an environment variable as well
```suggestion
	config["db_user"] = os.environ.get("FRAPPE_DB_USER") or config.get("db_user") or config.get("db_name")
```

blaggacao replied:
Good idea, thanks! :handshake: I wasn't confident myself since by the letter it'd be a "feature". :smiley:

blaggacao replied:
I force pushed to appease the semantic commit check.

akhilnarang reviewed (changes requested):
Minor thing, looks mostly fine. Will test in a bit.

blaggacao committed:
feat: db_user can be set by env varbiable

Co-authored-by: Akhil Narang <me@akhilnarang.dev>

blaggacao committed:
Merge remote-tracking branch 'origin/develop' into chore/cleanup-frappe-connect-2

blaggacao commented:
@mergify update

mergify[bot] committed:
Merge branch 'develop' into chore/cleanup-frappe-connect-2

akhilnarang commented on frappe/__init__.py:
```diff
@@ -293,11 +289,22 @@ def connect(
+			"Instead, explicitly invoke frappe.init(site) with the right config prior to calling frappe.connect(), if necessary."
+		)
+
+	assert db_name or local.conf.db_user, "site must be fully initialized, db_user missing"
```
Shouldn't we swap the order (both here and on line 307) - if someone has a user sent in the configuration that should be prioritized?

blaggacao replied:
Given that this removes the only known usage of this argument and deprecates it, I think you're right. That is a saner approach and only leaves this as escape hatch rather than an override.

I will change it shortly.

blaggacao replied:
Ok, it's done: https://github.com/frappe/frappe/pull/24432/commits/161bb4250871d23c346b40f1640562b054089c8b

blaggacao committed:
fix: deprioritize precedence of db_name argument to connect

blaggacao commented:
@akhilnarang in case you might miss it on the resolved conversation, I addressed your feedback above with https://github.com/frappe/frappe/pull/24432/commits/161bb4250871d23c346b40f1640562b054089c8b

akhilnarang commented on frappe/commands/utils.py:
```diff
@@ -772,12 +772,11 @@ def run_tests(
 			click.secho(f"bench --site {site} set-config allow_tests true", fg="green")
 			return
 
-		frappe.init(site=site)
```
Breaks `bench run-tests` since we need to access `frappe.flags` right below here
